static/app/components/metrics/metricSearchBar.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {MetricSearchBar} from 'sentry/components/metrics/metricSearchBar';
+
+describe('metricSearchBar', function () {
+  beforeEach(() => {
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/metrics/meta/',
+      body: [],
+    });
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/metrics/tags/',
+      body: [],
+    });
+    MockApiClient.addMockResponse({
+      method: 'POST',
+      url: '/organizations/org-slug/recent-searches/',
+      body: [],
+    });
+  });
+  it('does not allow illegal filters', async function () {
+    render(<MetricSearchBar onChange={() => undefined} />);
+    await screen.findByPlaceholderText('Filter by tags');
+    await userEvent.type(screen.getByPlaceholderText('Filter by tags'), 'potato:db');
+    expect(screen.getByTestId('search-autocomplete-item')).toHaveTextContent(
+      "The field potato isn't supported here."
+    );
+  });
+  it('does not allow insights filters when not using an insights mri', async function () {
+    render(<MetricSearchBar onChange={() => undefined} />);
+    await screen.findByPlaceholderText('Filter by tags');
+    await userEvent.type(screen.getByPlaceholderText('Filter by tags'), 'span.module:db');
+    expect(screen.getByTestId('search-autocomplete-item')).toHaveTextContent(
+      "The field span.module isn't supported here."
+    );
+  });
+  it('allows insights specific filters when using an insights mri', async function () {
+    const onChange = jest.fn();
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/metrics/tags/span.module/',
+      body: [],
+    });
+    render(
+      <MetricSearchBar onChange={onChange} mri="d:spans/exclusive_time@millisecond" />
+    );
+    await screen.findByPlaceholderText('Filter by tags');
+    await userEvent.type(screen.getByPlaceholderText('Filter by tags'), 'span.module:db');
+    await userEvent.keyboard('{enter}');
+    expect(onChange).toHaveBeenCalledWith('span.module:"db"');
```

Comment:
Thanks for adding tests for the metricsSearchBar 👏🚀

They look great, but I have a thought. In the last test, should we also check if the message "The field span.module isn't supported here." is not being rendered? Right now, we assume that if `onChange` is called, the message isn't shown. The same goes for the other tests -  should we also verify that `onChange` is not called?

Reply:
Updated the tests! I also realized that `MetricsSearchBar` uses either `SmartSearchBar` or `SearchQueryBuilder` depending on a feature flag, so I've updated to test both components!